libs/dyn/convert/struct_info.go

```diff
@@ -84,6 +87,16 @@ func buildStructInfo(typ reflect.Type) structInfo {
 			}
 
 			name, _, _ := strings.Cut(sf.Tag.Get("json"), ",")
+			if typ.Name() == "QualityMonitor" && name == "-" {
+				urlName, _, _ := strings.Cut(sf.Tag.Get("url"), ",")
+				if urlName == "" || urlName == "-" {
+					snake := matchFirstCap.ReplaceAllString(sf.Name, "${1}_${2}")
```

Comment:
Better  call it `CamelToSnakeCase` and move to `libs/textutil`

Reply:
Done.